plugins/account_history_plugin/include/eosio/account_history_plugin/account_control_history_object.hpp

```diff
@@ -39,7 +39,8 @@ using account_control_history_multi_index = chainbase::shared_multi_index_contai
       ordered_unique<tag<by_controlled_authority>,
          composite_key< account_control_history_object,
             member<account_control_history_object, account_name, &account_control_history_object::controlled_account>,
-            member<account_control_history_object, permission_name, &account_control_history_object::controlled_permission>
+            member<account_control_history_object, permission_name, &account_control_history_object::controlled_permission>,
+            member<account_control_history_object, account_name, &account_control_history_object::controlling_account>
```

Comment:
Is this an accidental commit? account_name is already part of the key at line 41.

Reply:
No, it wasn't accidentally.
If the index key of *by_controlled_authority* does not include the *controlling_account*, when a permission is changed to be controlled by *more than one account* an exception is raised and eosiod crash.

You can reproduce it with:
```bash
eosc set account permission test active '{"threshold" : 100, "keys" : [], "accounts" : [{"permission":{"account":"acc2","permission":"active"},"weight":50},{"permission":{"account":"acc1","permission":"active"},"weight":50}]}'
``` 

```
2237002ms thread-0   producer_plugin.cpp:209       block_production_loo ] Got exception while generating block:
13 St9exception: could not insert object, most likely a uniqueness constraint was violated
could not insert object, most likely a uniqueness constraint was violated: 
    {"trace.block":{"previous":"0000002741af33a5becd7ca0d874b8b9b04ac3b4aef3207a6f106213e6bdda1b","timestamp":"2018-02-02T00:37:17.000","transaction_mroot":"18a585ab1dc47946b33bd9ce07ebc7090b00346fde6083ff6d24a677104af19b","action_mroot":"1ae6af0fcef4829e531ea5a65673ba18fc1a0809cf915b0bd9c2c52ca38a54ff","block_mroot":"86194cf9b569c2fb851aeeb58f5f3e9c5a8a36d2ff2f9ef9631d88b590ee8161","producer":"initg","producer_signature":"EOSK7rHDjZ7U7SioB3thZT4jAUTJA2Hms5DJAxMv2EQGMB4SsFUZKdevwKsw68cG2MkPbr46Sz9UKo1kWYsKzyFvgau2uo6oP","regions":[{"region":0,"cycles_summary":[[{"read_locks":[],"write_locks":[{"account":"eosio","scope":"eosio.auth"}],"transactions":[{"status":"executed","id":"18a585ab1dc47946b33bd9ce07ebc7090b00346fde6083ff6d24a677104af19b"}]}]]}],"input_transactions":[{"expiration":"2018-02-02T00:37:46","region":0,"ref_block_num":39,"ref_block_prefix":2692533694,"actions":[{"account":"eosio","name":"updateauth","authorization":[{"actor":"test","permission":"active"}],"data":"000000000090b1ca00000000a8ed32320000000080ab26a76400000002000000000020103200000000a8ed32323200000000000010103200000000a8ed3232320000"}],"signatures":["EOSKjdgDzBgyhRtdpnRtNXqof1Jvkmhzcj3RBUCvcE6sYWvwepUgJKaGy8b7RhmRyCmA32dGDwn4c8mmHJtuZAn5KWPZGFevY"]}]},"what":"could not insert object, most likely a uniqueness constraint was violated"}
    thread-0  chain_controller.cpp:445 _finalize_block

    {"producer":"initg"}
    thread-0  chain_controller.cpp:514 _generate_block

    {"when":"2018-02-02T00:37:17.000"}
    thread-0  chain_controller.cpp:459 generate_block
2237002ms thread-0   producer_plugin.cpp:245       block_production_loo ] exception producing block
```

This also made me ask if we should be encolsing in a try/catch block the notifications to signal handlers in *chain_controller*

https://github.com/EOSIO/eos/blob/a407413e234ed966f3ce62b3edf574e653bdbe24/libraries/chain/chain_controller.cpp#L449

Reply:
Sorry, wasn't looking close enough to see that it was controlling instead of controlled.  It was changed from ordered_non_unique to ordered_unique when it was ported over and we are still limited with tests, thanks for catching that.